kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/OperatorChangeDetectionST.java

```diff
@@ -78,8 +73,43 @@ void shouldUpdateDeploymentWhenKafkaProxyIngressChanges(String namespace) {
         });
     }
 
-    private static String getChecksumFromAnnotation(HasMetadata entity) {
-        return KubernetesResourceUtil.getOrCreateAnnotations(entity).get("kroxylicious.io/referent-checksum");
+    @Test
+    void shouldRolloutDeploymentWhenFilterConfigurationChanges(String namespace) {
+        // Given
+        // @formatter:off
+        KafkaProtocolFilterBuilder arbitraryFilter = KroxyliciousFilterTemplates.baseFilterDeployment(namespace, "arbitrary-filter")
+                .withNewSpec()
+                    .withType("io.kroxylicious.proxy.filter.simpletransform.ProduceRequestTransformation")
+                    .withConfigTemplate(Map.of("findValue", "foo", "replacementValue", "bar"))
+                .endSpec();
+        // @formatter:on
+        KubeClient kubeClient = kubeClient(namespace);
+        resourceManager.createOrUpdateResourceWithWait(arbitraryFilter);
+        kroxylicious.deployPortIdentifiesNodeWithFilters(kafkaClusterName, List.of("arbitrary-filter"));
+
+        String originalChecksum = getInitialChecksum(namespace, kubeClient);
+
+        // @formatter:off
+        KafkaProtocolFilterBuilder updatedProtocolFilter = kubeClient.getClient().resources(KafkaProtocolFilter.class)
+                .inNamespace(namespace)
+                .withName("arbitrary-filter")
+                .get()
+                .edit()
+                    .editSpec()
+                    .withConfigTemplate(Map.of("findValue", "foo", "replacementValue", "updated"))
+                .endSpec();
+        // @formatter:on
+
+        // When
+        resourceManager.createOrUpdateResourceWithWait(updatedProtocolFilter);
+        LOGGER.info("Kafka proxy filter updated");
+
+        // Then
+        await().atMost(Duration.ofSeconds(90)).untilAsserted(() -> {
+            Deployment proxyDeployment = kubeClient.getDeployment(namespace, "simple");
+            OperatorAssertions.assertThat(proxyDeployment.getSpec().getTemplate().getMetadata()).hasAnnotationSatisfying("kroxylicious.io/referent-checksum",
```

Comment:
What if `proxyDeployment` is null? Is this already solved in this PR? I was going to investigate on this #2150

Reply:
If the deployment is null all bets are off as far as this test is concerned.

That said I haven't seen it fail locally so I haven't been trying to work out what's going wrong with the template deploy